python/tritonserver/_api/_response.py

```diff
@@ -29,293 +29,37 @@
-            iterator queue.
-        raise_on_error : bool
-            if True response errors will be raised as exceptions.
-
-        """
-
-        self._queue = queue.SimpleQueue()
-        self._user_queue = user_queue
-        self._complete = False
-        self._request = request
-        self._model = model
-        self._raise_on_error = raise_on_error
-
-    def __iter__(self) -> ResponseIterator:
-        """Return response iterator.
-
-        Returns
-        -------
-        ResponseIterator
-
-        Examples
-        --------
-
-        responses = server.model("test").infer(inputs={"fp16_input":numpy.array([[1]],dtype=numpy.float16)})
-        for response in responses:
-           print(nummpy.from_dlpack(response.outputs["fp16_output"]))
-
-
-        """
-
-        return self
-
-    def __next__(self):
-        """Returns the next response received for a request
-
-        Raises
-        ------
-        response.error
-            If raise_on_error is set to True, response errors are
-            raised as exceptions
-        StopIteration
-            Indicates all responses for a request have been received.
-            Final responses may or may not include outputs and must be
-            checked.
-
-        """
-
-        if self._complete:
-            raise StopIteration
-        response = self._queue.get()
-        self._complete = response.final
-        if response.error is not None and self._raise_on_error:
-            raise response.error
-        return response
-
-    def cancel(self):
-        """Cancel an inflight request
-
-        Cancels an in-flight request. Cancellation is handled on a
-        best effort basis and may not prevent execution of a request
-        if it is already started or completed.
-
-        See c:func:`TRITONSERVER_ServerInferenceRequestCancel`
-
-        Examples
-        --------
-
-        responses = server.model("test").infer(inputs={"text_input":["hello"]})
-
-        responses.cancel()
-
-        """
-        if self._request is not None:
-            self._request.cancel()
-
-    def _response_callback(self, response, flags, unused):
-        try:
-            if self._request is None:
-                raise InternalError("Response received after final response flag")
-
-            response = InferenceResponse._from_tritonserver_inference_response(
-                self._model, self._request, response, flags
-            )
-            self._queue.put(response)
-            if self._user_queue is not None:
-                self._user_queue.put(response)
-            if flags == TRITONSERVER_ResponseCompleteFlag.FINAL:
-                del self._request
-                self._request = None
-        except Exception as e:
-            message = f"Catastrophic failure in response callback: {e}"
-            LogMessage(LogLevel.ERROR, message)
-            # catastrophic failure
-            raise e from None
+DeviceOrMemoryType = (
```

Comment:
nit: this may be better defined in _memorybuffer.py and reused

Reply:
sure, unified the definition across apis. [Unify definition of DeviceOrMemoryType](https://github.com/triton-inference-server/core/pull/426/commits/bc6e7acac7fc021d128de5951f33ee920362da2e)